File: socialview-commons/src/com/hendraanggrian/appcompat/socialview/Hashtag.java
package com.hendraanggrian.appcompat.socialview;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * A data class of hashtag to be used with {@link com.hendraanggrian.widget.HashtagAdapter}.
 */
public class Hashtag implements Hashtagable {

    private final CharSequence hashtag;
    private final int count;

    public Hashtag(@NonNull CharSequence hashtag) {
        this(hashtag, -1);
    }

    public Hashtag(@NonNull CharSequence hashtag, int count) {
        this.hashtag = hashtag;
        this.count = count;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return obj instanceof Hashtag && ((Hashtag) obj).hashtag == hashtag;
    }

    @Override
    public int hashCode() {
        return hashtag.hashCode();
    }

    @NonNull
    @Override
    public String toString() {
        return hashtag.toString();
    }

    /**
     * {@inheritDoc}
     */
    @NonNull
    @Override
    public CharSequence getHashtag() {
        return hashtag;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public int getCount() {
        return count;
    }
}

File: socialview-suggestions/src/com/hendraanggrian/appcompat/socialview/Hashtag.kt
package com.hendraanggrian.appcompat.socialview

/** A data class of hashtag to be used with [com.hendraanggrian.widget.HashtagAdapter]. */
data class Hashtag @JvmOverloads constructor(
    override val hashtag: String,
    override val count: Int? = null

) : Hashtagable {

    override fun equals(other: Any?): Boolean = other != null &&
        other is Hashtag &&
        other.hashtag == hashtag

    override fun hashCode(): Int = hashtag.hashCode()

    override fun toString(): String = hashtag
}